streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -54,18 +55,19 @@
     private final PartitionGroup partitionGroup;
     private final PartitionGroup.RecordInfo recordInfo = new PartitionGroup.RecordInfo();
     private final PunctuationQueue punctuationQueue;
-    private final Map<TopicPartition, RecordQueue> partitionQueues;
 
     private final Map<TopicPartition, Long> consumedOffsets;
+    final Producer<byte[], byte[]> producer;
```

Comment:
For testing only. I'll try to make it private using your ideas from: https://github.com/apache/kafka/pull/2773#discussion_r109122317